File: lessons/01-Iterations/solution.py
def solution(N):
    binary = bin(N)[2:]
    
    arrayOfMax = []
    counters = 0

    for i in binary:
        if i == "0":
            counters += 1
        elif i == "1":
            arrayOfMax.append(counters)
            counters = 0

    print(max(arrayOfMax))


solution(8769)

File: lessons/01-Iterations/solution.js
function solution(N) {
    let binary = N.toString(2);
    const arrayofMax = [];
    let counts = 0;

    for(let i of binary) {
        if (i === "0") {
            counts++;
        } else {
            arrayofMax.push(counts);
            counts = 0;
        }
    }
    
    const result = Math.max(...arrayofMax);
    console.log(result);
    
}

solution(8769);